lib/llvm-backend/src/platform/unix.rs

```diff
@@ -68,3 +74,60 @@ extern "C" fn signal_trap_handler(
         throw_trap(2);
     }
 }
+
+pub unsafe fn alloc_memory(
+    size: usize,
+    protect: MemProtect,
+    ptr_out: &mut *mut u8,
+    size_out: &mut usize,
+) -> LLVMResult {
+    let size = round_up_to_page_size(size);
```

Comment:
Assuming the rust mmap follows POSIX mmap, neither the `addr` nor `length` arguments need to be page size aligned.

Reply:
This was same code as before, just moved into a new file.
Perhaps we can address this in a separate PR?